nilearn/_utils/testing.py

```diff
@@ -152,7 +152,7 @@ def write_tmp_imgs(*imgs, **kwargs):
     create_files = kwargs.get("create_files", True)
     use_wildcards = kwargs.get("use_wildcards", False)
 
-    prefix = "nilearn_"
+    prefix = "nilearn_{}_".format(os.getpid())
```

Comment:
you don't need to add the pid, mkstemp will take care of creating a filename that doesn't exist

Reply:
Thanks! Yes I got confused and that was a stupid attempt... :sweat: :smile:

Reply:
not at all! these things can get confusing indeed :smile: